commons-rest/src/main/java/io/tackle/commons/resources/query/QueryBuilder.java

```diff
@@ -1,35 +1,50 @@
 package io.tackle.commons.resources.query;
 
 import io.quarkus.hibernate.orm.panache.PanacheEntity;
+import io.tackle.commons.annotations.CheckType;
 import io.tackle.commons.annotations.Filterable;
 import io.tackle.commons.resources.ListFilteredResource;
 
+import javax.persistence.ElementCollection;
 import javax.persistence.ManyToMany;
 import javax.persistence.ManyToOne;
 import javax.persistence.OneToMany;
 import javax.persistence.OneToOne;
-import javax.ws.rs.WebApplicationException;
+import javax.ws.rs.BadRequestException;
 import javax.ws.rs.core.UriInfo;
 import java.lang.reflect.Field;
+import java.lang.reflect.ParameterizedType;
+import java.lang.reflect.Type;
 import java.util.ArrayList;
 import java.util.Arrays;
 import java.util.HashMap;
 import java.util.List;
 import java.util.Map;
 import java.util.concurrent.ThreadLocalRandom;
 import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.function.Function;
 import java.util.stream.Collectors;
 
-import static javax.ws.rs.core.Response.Status.BAD_REQUEST;
-
 // https://github.com/quarkusio/quarkus/issues/15088#issuecomment-783454416
 // Class to cover the need of generating queries on our own
 public class QueryBuilder<ENTITY extends PanacheEntity> implements QueryUriInfo<ENTITY>, QueryBuild<ENTITY> {
 
     public static final String DEFAULT_SQL_ROOT_TABLE_ALIAS = "table";
+    private static final QueryParameterGenerators EQUAL_QUERY_PARAMETER_GENERATORS = new QueryParameterGenerators();
     private final Class<ENTITY> panacheEntity;
     private UriInfo uriInfo;
-    private final Map<String, Boolean> filterableFields;
+    private final Map<String, Field> filterableFields;
+
+    static {
+        EQUAL_QUERY_PARAMETER_GENERATORS.putQueryParameterGenerator(Long.class, value -> {
+            try {
+                return Long.valueOf(value);
+            } catch (NumberFormatException e) {
+                throw new BadRequestException("Malformed filter value");
+            }
+        });
+        EQUAL_QUERY_PARAMETER_GENERATORS.putQueryParameterGenerator(String.class, Function.identity());
```

Comment:
The overall change of this PR is to introduce the "EQUAL" operator; the "EQUAL" operator can be applied against all primitive java types like: "boolean, integer, double, float, string", in short words "Numbers, boolean, and string". I wonder if the current PR is only valid for fields whose type is "String" or "Float"? since this is what we need right now I'm happy with the code as it is but I don't know how much effort it might require to include the other java types, if it is a matter of minutes could we consider adding the other types too? otherwise let's create an issue in order to enhance the current code in the future and not block the other PRs

Reply:
I think it's a good idea @carlosthe19916 to have the basic types covered in this PR so that they're done and no further changes are required.